From 8836b9d243444031b6396d39d345f2f83b5fa6a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Spintzyk?= <lukasz.spintzyk@synaptics.com>
Date: Thu, 22 Jul 2021 13:12:05 +0200
Subject: [PATCH 2/2] modesetting: unflip not possible when glamor is not set

This is fixing crashes of xfce when running under qemu

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 hw/xfree86/drivers/modesetting/present.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/xfree86/drivers/modesetting/present.c b/hw/xfree86/drivers/modesetting/present.c
index f0a0e2704..916a15303 100644
--- a/hw/xfree86/drivers/modesetting/present.c
+++ b/hw/xfree86/drivers/modesetting/present.c
@@ -272,6 +272,9 @@ ms_present_check_unflip(RRCrtcPtr crtc,
         pixmap->devKind != drmmode_bo_get_pitch(&ms->drmmode.front_bo))
         return FALSE;
 
+    if (!ms->drmmode.glamor)
+        return FALSE;
+
 #ifdef GBM_BO_WITH_MODIFIERS
     /* Check if buffer format/modifier is supported by all active CRTCs */
     gbm = ms->glamor.gbm_bo_from_pixmap(screen, pixmap);
-- 
2.32.0

